app/gui/view/graph-editor/src/component/node/input/widget/hierarchy.rs

```diff
@@ -57,14 +48,21 @@ impl SpanWidget for Widget {
         let display_object = object::Instance::new_named("widget::Hierarchy");
         display_object.use_auto_layout();
         display_object.set_children_alignment_left_center().justify_content_center_y();
-        Self { display_object }
+        Self { display_object, children_vec: default() }
     }
 
     fn configure(&mut self, _: &Config, ctx: ConfigContext) {
         let child_level = ctx.info.nesting_level.next_if(ctx.span_node.is_argument());
-        let children_iter = ctx.span_node.children_iter();
-        let children =
-            children_iter.map(|node| ctx.builder.child_widget(node, child_level).root_object);
-        self.display_object.replace_children(&children.collect::<CollectedChildren>());
+        let is_primary = ctx.info.nesting_level.is_primary();
+        let separator_config = Configuration::inert(super::separator::Config.into());
+        let child_config = is_primary.then_some(&separator_config);
```

Comment:
Basically, if this is a top-level (primary) `hierarchy` widget, then it will request its children widgets to be separators. Otherwise, the default is used.